# Uncovering Insights into Maternal Health: Analyzing Trends, Disparities, and Risk Factors

## Importing the required libraries and the dataset 

In [3]:
import sqlite3
import pandas as pd

# Path to your CSV file
csv_file_path = 'dataset/MCH_Indicators_2016_2022.csv'

# Replace 'your_database_name.db' with your desired file name
db_path = 'Maternal_Indicators.db'


# Connect to the SQLite database
conn = sqlite3.connect(db_path)

# Read CSV into DataFrame
df = pd.read_csv(csv_file_path)

# Write DataFrame to SQL Table, assume table name as 'Maternal_Indicators'
df.to_sql('Maternal_Indicators', con=conn, if_exists='replace', index=False)

# Close the connection
conn.close()

In [4]:
%load_ext sql

In [5]:
%sql sqlite:///Maternal_Indicators.db

## Exploring the Dataset 

In [6]:
%sql SELECT * FROM Maternal_Indicators LIMIT 10;

 * sqlite:///Maternal_Indicators.db
Done.


[('Alaska', 2016, None, 1165, 435, 40.5, 37.0, 44.0, 1144, 34, 2.3, 1.5, 3.5, 1144, 267, 23.4, 20.5, 26.6, 1144, 289, 24.6, 21.7, 27.8, 1158, 330, 25.7, 22.9, 28.8, 1160, 162, 11.0, 9.2, 13.1, 1158, 213, 14.8, 12.7, 17.1, 1156, 50, 5.3, 3.8, 7.3, 1156, 15, 1.1, 0.6, 2.1, 1147, 39, 4.0, 2.7, 5.9, 1152, 43, 2.8, 1.9, 4.1, 1148, 50, 2.7, 1.9, 3.9, 1142, 39, 2.2, 1.5, 3.4, 1172, 160, 12.0, 10.0, 14.4, 1165, 181, 13.2, 11.1, 15.8, 1128, 155, 14.1, 11.7, 16.8, 1163, 724, 65.4, 62.1, 68.6, 1142, 964, 84.7, 81.9, 87.1, 1143, 696, 57.6, 53.9, 61.1, 1137, 968, 86.4, 83.9, 88.6, 1153, 190, 17.2, 14.6, 20.2, 1153, 59, 4.8, 3.5, 6.5, 1153, 212, 16.5, 14.1, 19.1, 1153, 692, 61.5, 58.0, 64.9, 1118, 872, 79.9, 76.9, 82.6, 1118, 135, 12.6, 10.3, 15.3, 1118, 278, 23.8, 20.9, 27.1, 1118, 230, 20.4, 17.6, 23.5, 1118, 229, 23.0, 20.0, 26.3, 1166, 521, 47.7, 44.2, 51.3, 1143, 611, 59.1, 55.8, 62.3, 1143, 304, 23.2, 20.6, 26.1, 1143, 228, 17.7, 15.2, 20.4),
 ('Arkansas', 2016, 17.0, 901, 298, 35.9, 31.4, 40.7, 864, 42, 3.8, 2.3, 6.0, 864, 203, 25.6, 21.5, 30.1, 864, 270, 28.9, 24.7, 33.6, 878, 246, 24.3, 20.4, 28.7, 880, 155, 15.7, 12.5, 19.6, 878, 201, 19.9, 16.3, 24.1, 884, 61, 8.3, 5.9, 11.6, 883, 28, 3.4, 2.0, 5.8, 883, 40, 4.4, 2.8, 7.0, 884, 19, 3.2, 1.8, 5.5, 880, 52, 6.0, 4.0, 8.7, 881, 28, 2.1, 1.1, 4.0, 900, 170, 19.0, 15.5, 23.1, 880, 187, 20.8, 17.1, 25.0, 884, 205, 19.2, 15.7, 23.3, 895, 491, 56.4, 51.5, 61.2, 886, 698, 80.1, 75.9, 83.8, 888, 483, 61.0, 56.2, 65.7, 875, 754, 89.9, 86.6, 92.5, 894, 198, 21.5, 17.7, 25.8, 894, 114, 9.8, 7.2, 13.0, 894, 190, 17.7, 14.3, 21.6, 894, 392, 51.1, 46.2, 56.0, 872, 704, 83.4, 79.4, 86.7, 872, 155, 14.7, 11.6, 18.5, 872, 110, 15.5, 12.2, 19.5, 872, 299, 34.8, 30.2, 39.7, 872, 140, 18.4, 14.8, 22.6, 890, 283, 36.2, 31.6, 41.1, 785, 468, 64.9, 59.7, 69.7, 785, 247, 24.2, 20.0, 28.9, 785, 70, 11.0, 8.0, 14.8),
 ('Colorado', 2016, 16.0, 1619, 753, 43.2, 39.9, 46.7, 1528, 63, 3.4, 2.4, 4.9, 1528, 325, 23.9, 21.0, 27.1, 1528, 274, 17.3, 14.7, 20.2, 1617, 262, 14.7, 12.4, 17.2, 1618, 116, 6.4, 5.0, 8.3, 1617, 175, 10.0, 8.1, 12.2, 1610, 61, 2.7, 1.8, 3.9, 1610, 17, 1.1, 0.6, 2.3, 1585, 77, 5.4, 4.1, 7.2, 1614, 74, 4.3, 3.2, 5.8, 1584, 40, 2.8, 1.8, 4.5, 1586, 29, 2.1, 1.2, 3.7, 1606, 208, 12.1, 10.0, 14.5, 1603, 208, 12.1, 10.0, 14.4, 1606, 185, 9.9, 8.0, 12.1, 1602, 1179, 69.0, 65.6, 72.3, 1592, 1425, 90.4, 88.0, 92.4, 1604, 1058, 66.7, 63.4, 69.8, 1611, 1476, 91.9, 89.7, 93.6, 1600, 271, 18.5, 15.9, 21.4, 1600, 68, 4.3, 3.0, 6.2, 1600, 234, 16.0, 13.5, 18.8, 1600, 1027, 61.2, 57.7, 64.6, 1586, 1258, 77.9, 74.8, 80.7, 1586, 193, 9.9, 8.1, 11.9, 1586, 352, 22.6, 19.8, 25.7, 1586, 347, 21.6, 18.8, 24.6, 1586, 366, 23.9, 21.0, 27.0, 1620, 768, 46.8, 43.4, 50.3, 1538, 1042, 66.3, 62.7, 69.6, 1538, 359, 24.1, 21.1, 27.3, 1538, 132, 9.5, 7.5, 12.1),
 ('Connecticut', 2016, None, 1286, 543, 47.8, 44.5, 51.1, 1232, 46, 3.6, 2.5, 5.1, 1232, 310, 24.6, 21.7, 27.7, 1232, 320, 26.1, 23.2, 29.3, 1269, 164, 16.5, 13.9, 19.4, 1271, 59, 5.5, 4.1, 7.5, 1271, 106, 9.4, 7.5, 11.7, 1276, 28, 2.6, 1.7, 4.0, 1280, 5, 0.3, 0.1, 0.7, 1274, 75, 5.1, 3.9, 6.6, 1272, 20, 2.1, 1.3, 3.6, 1278, 31, 2.2, 1.4, 3.4, 1275, 30, 2.3, 1.5, 3.7, 1284, 104, 8.9, 7.1, 11.1, 1266, 102, 8.7, 6.9, 10.9, 1265, 155, 10.5, 8.6, 12.6, 1280, 931, 78.7, 76.2, 81.0, 1272, 1150, 92.1, 90.3, 93.6, 1272, 882, 70.4, 67.2, 73.4, 1269, 1199, 95.4, 93.9, 96.5, 1279, 237, 16.9, 14.7, 19.5, 1279, 75, 4.7, 3.6, 6.2, 1279, 204, 14.6, 12.4, 17.2, 1279, 763, 63.7, 60.5, 66.8, 1256, 943, 75.2, 72.0, 78.1, 1256, 121, 9.5, 7.7, 11.7, 1256, 209, 16.9, 14.5, 19.6, 1256, 303, 25.1, 22.2, 28.2, 1256, 310, 23.7, 20.9, 26.7, 1284, 672, 58.8, 55.5, 62.0, 1263, 797, 67.4, 64.5, 70.1, 1263, 317, 23.2, 20.6, 26.0, 1263, 149, 9.4, 8.0, 11.1),
 ('Delaware', 2016, None, 928, 405, 41.8, 38.5, 45.1, 876, 29, 2.7, 1.8, 4.0, 876, 231, 26.9, 23.9, 30.1, 876, 237, 26.9, 24.0, 30.1, 923, 188, 20.1, 17.5, 22.9, 925, 95, 9.0, 7.3, 11.0, 925, 131, 13.0, 10.9, 15.5, 921, 34, 3.7, 2.7, 5.3, 92

## Dataset Overview

The Maternal_Indicators dataset contains information related to maternal health, including mortality rates, health indicators, and risk factors. The data is collected from various sites and spans multiple years, providing insights into trends and disparities in maternal health outcomes.

source of the dataset can be found on [Kaggle](https://www.kaggle.com/datasets/neharana404/maternal-indicators-in-us-states2016-2021)

## Prenatal Care and Behaviors:

### Early Prenatal Care Access

In [7]:
%%sql

SELECT 
    Year,
    "Began prenatal care in 1st trimester - Weighted %" AS "Early Prenatal Care %"
FROM 
    Maternal_Indicators
GROUP BY 
    "Year"
ORDER BY 
    "Year";


 * sqlite:///Maternal_Indicators.db
Done.


[(2016, 84.7),
 (2017, 83.0),
 (2018, 83.2),
 (2019, 83.0),
 (2020, 86.3),
 (2021, 84.3)]

- The percentage of women accessing early prenatal care shows minor fluctuations from year to year between 2016 and 2021, with a notable peak in 2020 

### Impact of Multivitamin Use on Maternal Mortality Rates

In [8]:
%%sql

SELECT 
    CAST(REPLACE("Multivitamin use ≥4 times a week during the month before pregnancy - Weighted %", '%', '') AS INTEGER) AS "Multivitamin Use %",
    CAST(AVG(CAST("Deaths/100000" AS REAL)) + 0.999 AS INTEGER) AS "Average Maternal Mortality Rate"
FROM 
    Maternal_Indicators
GROUP BY 
    "Multivitamin Use %"
ORDER BY 
    "Multivitamin Use %" DESC;

 * sqlite:///Maternal_Indicators.db
Done.


[(58, 15),
 (56, 0),
 (55, 31),
 (54, None),
 (53, 10),
 (52, 23),
 (51, 23),
 (50, 27),
 (49, None),
 (48, 17),
 (47, 31),
 (46, 35),
 (45, 30),
 (44, 33),
 (43, 32),
 (42, 23),
 (41, 29),
 (40, 41),
 (39, 21),
 (38, 30),
 (37, 26),
 (36, 17),
 (35, 36),
 (34, 29),
 (33, 54),
 (32, 31),
 (31, 34),
 (30, 74),
 (28, 29),
 (27, None),
 (22, None),
 (21, None)]

- There seems to be a potential benefit in some cases at higher multivitamin usage rates, but this is not universally true across all the data.

- At the highest reported use rates (58% and 56%), the maternal mortality rates are relatively low (15 and 0 per 100,000, respectively), but there seems to be a potential benefit in some cases at higher multivitamin usage rates, but this is not universally true across all the data.


## Postpartum Care:

### State-by-State Postpartum Checkup Rates

In [9]:
%%sql

SELECT 
    "Site Name",
    ROUND(AVG("Had maternal postpartum checkup - Weighted %"), 0) AS "Postpartum Checkup %"
FROM 
    Maternal_Indicators
GROUP BY
    "Site Name"
ORDER BY 
    "Postpartum Checkup %" DESC;



 * sqlite:///Maternal_Indicators.db
Done.


[('Rhode Island', 94.0),
 ('New Hampshire', 94.0),
 ('Connecticut', 94.0),
 ('Vermont', 93.0),
 ('Maine', 93.0),
 ('Iowa', 93.0),
 ('Illinois', 93.0),
 ('Wyoming', 92.0),
 ('Oregon', 92.0),
 ('North Carolina', 92.0),
 ('Massachusetts', 92.0),
 ('Alabama', 92.0),
 ('Wisconsin', 91.0),
 ('Washington', 91.0),
 ('Utah', 91.0),
 ('New York State', 91.0),
 ('Nebraska', 91.0),
 ('Montana', 91.0),
 ('Minnesota', 91.0),
 ('Kansas', 91.0),
 ('Hawaii', 91.0),
 ('Colorado', 91.0),
 ('Virginia', 90.0),
 ('South Dakota', 90.0),
 ('North Dakota', 90.0),
 ('Maryland', 90.0),
 ('Georgia', 90.0),
 ('District of Columbia', 90.0),
 ('West Virginia', 89.0),
 ('Tennessee', 89.0),
 ('Pennsylvania', 89.0),
 ('New Jersey', 89.0),
 ('Michigan', 89.0),
 ('Indiana', 89.0),
 ('Florida', 89.0),
 ('Arkansas', 89.0),
 ('Oklahoma', 88.0),
 ('Missouri', 88.0),
 ('Mississippi', 88.0),
 ('Kentucky', 88.0),
 ('Delaware', 88.0),
 ('Texas', 87.0),
 ('New Mexico', 87.0),
 ('Louisiana', 87.0),
 ('Alaska', 87.0),
 ('Arizona', 85.0),
 ('Puerto Rico', 78.0)]

- High Engagement in Northern States: States like Rhode Island, New Hampshire, and Connecticut lead with the highest rates of postpartum checkups, each at 94%.

### Correlation Between Prenatal Care Initiation and Postpartum Depressive Symptoms

In [10]:
%%sql

SELECT 
    ROUND("Began prenatal care in 1st trimester - Weighted %", 0) AS "Prenatal Care Initiation %",
    AVG(ROUND("Self-reported postpartum depressive symptoms‡ - Weighted %", 0)) AS "Average Postpartum Depressive Symptoms %"
FROM 
    Maternal_Indicators
GROUP BY 
    ROUND("Began prenatal care in 1st trimester - Weighted %", 0)
ORDER BY 
    "Prenatal Care Initiation %" DESC;



 * sqlite:///Maternal_Indicators.db
Done.


[(95.0, 11.333333333333334),
 (94.0, 10.8),
 (93.0, 12.0),
 (92.0, 11.25),
 (91.0, 11.933333333333334),
 (90.0, 12.851851851851851),
 (89.0, 12.705882352941176),
 (88.0, 12.787878787878787),
 (87.0, 13.96551724137931),
 (86.0, 13.620689655172415),
 (85.0, 14.0),
 (84.0, 15.090909090909092),
 (83.0, 17.153846153846153),
 (82.0, 15.5),
 (81.0, 11.0),
 (80.0, 14.666666666666666),
 (79.0, 11.0)]

- Higher rates of prenatal care initiation (around 95% to 91%) do not consistently correspond with lower average rates of postpartum depressive symptoms, suggesting that early prenatal care alone may not significantly influence postpartum depression rates.


### Contraceptive use among postpartum women across different years

In [11]:
%%sql

SELECT
    Year,
    ROUND("Use of any postpartum contraception§‖ - Weighted %", 0) AS "Postpartum Contraceptive Use %",
    ROUND("Highly effective contraceptive method: Male or female sterilization - Weighted %", 0) AS "Sterilization %",
    ROUND("Highly effective contraceptive method: Long-acting reversible contraceptive method¶ -Weighted %", 0) AS "LARC %",
    ROUND("Moderately effective contraceptive methods¶ - Weighted %", 0) AS "Moderately Effective Methods %",
    ROUND("Least effective contraceptive methods¶ - Weighted %", 0) AS "Least Effective Methods %"
FROM
    Maternal_Indicators
GROUP BY
    Year;

 * sqlite:///Maternal_Indicators.db
Done.


[(2016, 80.0, 13.0, 24.0, 20.0, 23.0),
 (2017, 81.0, 18.0, 16.0, 31.0, 16.0),
 (2018, 78.0, 13.0, 14.0, 32.0, 19.0),
 (2019, 83.0, 18.0, 15.0, 32.0, 19.0),
 (2020, 79.0, 16.0, 13.0, 32.0, 18.0),
 (2021, 77.0, 15.0, 14.0, 31.0, 17.0)]

- The percentage of postpartum contraceptive use shows fluctuations over the years, with a notable decrease from 80% in 2016 to 77% in 2021.

- Use of long-acting reversible contraceptives (LARC) and sterilization varies, but the use of moderately effective methods has generally increased, peaking in 2019 and 2021 at 32%.


## Health Disparities:

### Variations in prenatal care access across geographic regions

In [12]:
%%sql

SELECT
    "Site Name" AS Site,
    ROUND(AVG("Began prenatal care in 1st trimester - Weighted %"), 0) AS "Prenatal Care Initiation %"
FROM
    Maternal_Indicators
GROUP BY
    "Site Name"
ORDER BY
    "Prenatal Care Initiation %" DESC;


 * sqlite:///Maternal_Indicators.db
Done.


[('Vermont', 93.0),
 ('New Hampshire', 93.0),
 ('Rhode Island', 92.0),
 ('Connecticut', 92.0),
 ('New York State', 91.0),
 ('Massachusetts', 91.0),
 ('Washington', 90.0),
 ('Oregon', 90.0),
 ('Maine', 90.0),
 ('Colorado', 90.0),
 ('West Virginia', 89.0),
 ('Utah', 89.0),
 ('Nebraska', 89.0),
 ('Minnesota', 89.0),
 ('Illinois', 89.0),
 ('Wisconsin', 88.0),
 ('Puerto Rico', 88.0),
 ('North Dakota', 88.0),
 ('Montana', 88.0),
 ('Louisiana', 88.0),
 ('Kentucky', 88.0),
 ('Kansas', 88.0),
 ('Iowa', 88.0),
 ('Virginia', 87.0),
 ('Pennsylvania', 87.0),
 ('New Jersey', 87.0),
 ('Missouri', 87.0),
 ('Michigan', 87.0),
 ('Hawaii', 87.0),
 ('Georgia', 87.0),
 ('Wyoming', 86.0),
 ('Tennessee', 86.0),
 ('South Dakota', 86.0),
 ('North Carolina', 86.0),
 ('District of Columbia', 86.0),
 ('New Mexico', 85.0),
 ('Maryland', 85.0),
 ('Florida', 85.0),
 ('Arizona', 85.0),
 ('Alaska', 85.0),
 ('Oklahoma', 84.0),
 ('Mississippi', 84.0),
 ('Arkansas', 84.0),
 ('Alabama', 84.0),
 ('Indiana', 82.0),
 ('Delaware', 82.0),
 ('Texas', 80.0)]

- Vermont and New Hampshire lead with the highest rates of prenatal care initiation at 93%, indicating strong early engagement in maternal healthcare in these states.

- The data shows a range from high initiation rates in the Northeast to lower rates in states like Texas at 80%

### Calculate the range and average of prenatal care initiation rates across all sites

In [13]:
%%sql

SELECT
    ROUND(MAX("Began prenatal care in 1st trimester - Weighted %"), 0) AS "Max Prenatal Care Initiation %",
    ROUND(MIN("Began prenatal care in 1st trimester - Weighted %"), 0) AS "Min Prenatal Care Initiation %",
    ROUND(AVG("Began prenatal care in 1st trimester - Weighted %"), 0) AS "Average Prenatal Care Initiation %"
FROM
    Maternal_Indicators;


 * sqlite:///Maternal_Indicators.db
Done.


[(95.0, 79.0, 88.0)]

- The maximum prenatal care initiation rate across the dataset is 95%, while the minimum is significantly lower at 79%, indicating a wide disparity in early prenatal care engagement

### Geographic Distribution of Postpartum Depression Rates Across the U.S.

In [14]:
%%sql

SELECT
    "Site Name" AS Site,
    ROUND(AVG("Self-reported postpartum depressive symptoms‡ - Weighted %"), 1) AS "Postpartum Depression %"
FROM
    Maternal_Indicators
GROUP BY
    "Site Name"
ORDER BY
    "Postpartum Depression %" DESC;

 * sqlite:///Maternal_Indicators.db
Done.


[('Mississippi', 22.0),
 ('Arkansas', 20.4),
 ('Alabama', 20.0),
 ('Indiana', 18.1),
 ('Oklahoma', 15.4),
 ('Utah', 15.3),
 ('West Virginia', 15.2),
 ('Michigan', 15.2),
 ('Texas', 15.1),
 ('Louisiana', 15.1),
 ('Tennessee', 14.6),
 ('Montana', 14.6),
 ('Wyoming', 14.5),
 ('Maryland', 14.3),
 ('Florida', 14.3),
 ('Kansas', 14.0),
 ('New Mexico', 13.9),
 ('Kentucky', 13.9),
 ('Arizona', 13.7),
 ('Alaska', 13.7),
 ('Missouri', 13.6),
 ('District of Columbia', 13.4),
 ('North Dakota', 13.1),
 ('New Hampshire', 13.0),
 ('Virginia', 12.8),
 ('Hawaii', 12.8),
 ('South Dakota', 12.7),
 ('Rhode Island', 12.5),
 ('Puerto Rico', 12.3),
 ('Wisconsin', 12.2),
 ('Oregon', 12.2),
 ('Pennsylvania', 12.0),
 ('New York State', 12.0),
 ('Washington', 11.9),
 ('Georgia', 11.9),
 ('Nebraska', 11.7),
 ('Minnesota', 11.7),
 ('Maine', 11.5),
 ('North Carolina', 11.4),
 ('New Jersey', 11.4),
 ('Connecticut', 11.4),
 ('Colorado', 11.2),
 ('Delaware', 11.0),
 ('Vermont', 10.9),
 ('Massachusetts', 10.5),
 ('Illinois', 10.3),
 ('Iowa', 9.4)]

- The highest rates of postpartum depression are found in southern states, with Mississippi, Arkansas, and Alabama leading, showing rates of 22.0%, 20.4%, and 20.0% respectively.

- States like Iowa, Illinois, and Massachusetts report significantly lower rates of postpartum depression, with the lowest being Iowa at 9.4%.

## Insurance 

In [15]:
%%sql
SELECT
    "Site Name" AS State,
    Year,
    ROUND("Private insurance - Weighted %", 0) AS "Private Insurance %",
    ROUND("Medicaid or CHIP/SCHIP - Weighted %", 0) AS "Medicaid or CHIP/SCHIP %",
    ROUND("No insurance - Weighted %", 0) AS "No Insurance %",
    CASE
        WHEN "Private insurance - Weighted %" > COALESCE("Medicaid or CHIP/SCHIP - Weighted %", 0) AND "Private insurance - Weighted %" > COALESCE("No insurance - Weighted %", 0) THEN 'Private Insurance'
        WHEN "Medicaid or CHIP/SCHIP - Weighted %" > COALESCE("Private insurance - Weighted %", 0) AND "Medicaid or CHIP/SCHIP - Weighted %" > COALESCE("No insurance - Weighted %", 0) THEN 'Medicaid or CHIP/SCHIP'
        WHEN "No insurance - Weighted %" > COALESCE("Private insurance - Weighted %", 0) AND "No insurance - Weighted %" > COALESCE("Medicaid or CHIP/SCHIP - Weighted %", 0) THEN 'No Insurance'
        ELSE 'Unknown'
    END AS "Most Used Insurance"
FROM
    Maternal_Indicators
    
ORDER BY
    State, Year;


 * sqlite:///Maternal_Indicators.db
Done.


[('Alabama', 2017, 60.0, 20.0, 21.0, 'Private Insurance'),
 ('Alabama', 2018, 59.0, 24.0, 17.0, 'Private Insurance'),
 ('Alabama', 2019, 62.0, 22.0, 16.0, 'Private Insurance'),
 ('Alabama', 2020, 62.0, 22.0, 17.0, 'Private Insurance'),
 ('Alabama', 2021, 58.0, 23.0, 19.0, 'Private Insurance'),
 ('Alaska', 2016, 59.0, 23.0, 18.0, 'Private Insurance'),
 ('Alaska', 2017, 62.0, 23.0, 15.0, 'Private Insurance'),
 ('Alaska', 2018, 61.0, 27.0, 12.0, 'Private Insurance'),
 ('Alaska', 2019, 60.0, 29.0, 11.0, 'Private Insurance'),
 ('Alaska', 2020, 58.0, 30.0, 11.0, 'Private Insurance'),
 ('Arizona', 2020, 52.0, 34.0, 15.0, 'Private Insurance'),
 ('Arkansas', 2016, 65.0, 24.0, 11.0, 'Private Insurance'),
 ('Arkansas', 2018, 59.0, 29.0, 12.0, 'Private Insurance'),
 ('Arkansas', 2019, 60.0, 28.0, 12.0, 'Private Insurance'),
 ('Arkansas', 2020, 56.0, 30.0, 13.0, 'Private Insurance'),
 ('Arkansas', 2021, 55.0, 32.0, 14.0, 'Private Insurance'),
 ('Colorado', 2016, 66.0, 24.0, 10.0, 'Private Insurance'),
 ('Colorado', 2017, 67.0, 22.0, 10.0, 'Private Insurance'),
 ('Colorado', 2018, 63.0, 26.0, 11.0, 'Private Insurance'),
 ('Colorado', 2019, 66.0, 24.0, 10.0, 'Private Insurance'),
 ('Colorado', 2020, 67.0, 21.0, 12.0, 'Private Insurance'),
 ('Colorado', 2021, 65.0, 22.0, 13.0, 'Private Insurance'),
 ('Connecticut', 2016, 67.0, 23.0, 9.0, 'Private Insurance'),
 ('Connecticut', 2017, 65.0, 26.0, 9.0, 'Private Insurance'),
 ('Connecticut', 2018, 62.0, 28.0, 10.0, 'Private Insurance'),
 ('Connecticut', 2019, 66.0, 25.0, 10.0, 'Private Insurance'),
 ('Connecticut', 2020, 64.0, 27.0, 9.0, 'Private Insurance'),
 ('Connecticut', 2021, 65.0, 26.0, 9.0, 'Private Insurance'),
 ('Delaware', 2016, 58.0, 31.0, 9.0, 'Private Insurance'),
 ('Delaware', 2017, 58.0, 32.0, 9.0, 'Private Insurance'),
 ('Delaware', 2018, 58.0, 30.0, 11.0, 'Private Insurance'),
 ('Delaware', 2019, 58.0, 29.0, 11.0, 'Private Insurance'),
 ('Delaware', 2020, 57.0, 32.0, 10.0, 'Private Insurance'),
 ('Delaware', 2021, 58.0, 32.0, 10.0, 'Private Insurance'),
 ('District of Columbia', 2018, 55.0, 34.0, 4.0, 'Private Insurance'),
 ('District of Columbia', 2019, 54.0, 34.0, 6.0, 'Private Insurance'),
 ('District of Columbia', 2020, 57.0, 34.0, 4.0, 'Private Insurance'),
 ('District of Columbia', 2021, 57.0, 32.0, 5.0, 'Private Insurance'),
 ('Florida', 2019, 63.0, 17.0, 20.0, 'Private Insurance'),
 ('Florida', 2020, 59.0, 20.0, 21.0, 'Private Insurance'),
 ('Georgia', 2017, 62.0, 13.0, 25.0, 'Private Insurance'),
 ('Georgia', 2018, 56.0, 17.0, 27.0, 'Private Insurance'),
 ('Georgia', 2019, 54.0, 17.0, 29.0, 'Private Insurance'),
 ('Georgia', 2020, 51.0, 18.0, 31.0, 'Private Insurance'),
 ('Georgia', 2021, 58.0, 18.0, 23.0, 'Private Insurance'),
 ('Hawaii', 2016, 72.0, 24.0, 4.0, 'Private Insurance'),
 ('Hawaii', 2019, 69.0, 27.0, 4.0, 'Private Insurance'),
 ('Hawaii', 2020, 72.0, 24.0, 5.0, 'Private Insurance'),
 ('Hawaii', 2021, 72.0, 25.0, 3.0, 'Private Insurance'),
 ('Illinois', 2016, 62.0, 25.0, 12.0, 'Private Insurance'),
 ('Illinois', 2017, 60.0, 26.0, 14.0, 'Private Insurance'),
 ('Illinois', 2018, 63.0, 25.0, 11.0, 'Private Insurance'),
 ('Illinois', 2019, 61.0, 25.0, 14.0, 'Private Insurance'),
 ('Illinois', 2020, 63.0, 25.0, 12.0, 'Private Insurance'),
 ('Illinois', 2021, 61.0, 25.0, 14.0, 'Private Insurance'),
 ('Indiana', 2018, 56.0, 13.0, 16.0, 'Private Insurance'),
 ('Iowa', 2016, 66.0, 25.0, 9.0, 'Private Insurance'),
 ('Iowa', 2017, 68.0, 21.0, 9.0, 'Private Insurance'),
 ('Iowa', 2018, 71.0, 23.0, 6.0, 'Private Insurance'),
 ('Iowa', 2019, 72.0, 21.0, 7.0, 'Private Insurance'),
 ('Iowa', 2020, 69.0, 21.0, 9.0, 'Private Insurance'),
 ('Kansas', 2017, 73.0, 12.0, 14.0, 'Private Insurance'),
 ('Kansas', 2018, 69.0, 13.0, 18.0, 'Private Insurance'),
 ('Kansas', 2019, 68.0, 15.0, 18.0, 'Private Insurance'),
 ('Kansas', 2020, 69.0, 14.0, 18.0, 'Private Insurance'),
 ('Kansas', 2021, 71.0, 15.0, 13.0, 'Private Insurance'),
 ('Kentucky', 2017, 58.0, 35.0, 7.0, 'Private Insurance

## Smoking


### Trend of smoking before pregnancy over the years


In [16]:
%%sql
SELECT
    Year,
    AVG(ROUND("Any cigarette smoking during the 3 months before pregnancy - Weighted %", 0)) AS "Smoking Before Pregnancy %"
FROM
    Maternal_Indicators
GROUP BY
    Year;

 * sqlite:///Maternal_Indicators.db
Done.


[(2016, 19.5),
 (2017, 19.485714285714284),
 (2018, 18.404761904761905),
 (2019, 16.714285714285715),
 (2020, 15.512195121951219),
 (2021, 13.166666666666666)]

- There has been a consistent decrease in the percentage of women who reported smoking before pregnancy, from 19.5% in 2016 to 13.2% in 2021.

### Smoking rates before, during, and after pregnancy

In [17]:
%%sql

SELECT
    Year,
    ROUND(AVG(CASE WHEN "Any cigarette smoking during the 3 months before pregnancy - Weighted %" IS NOT NULL THEN "Any cigarette smoking during the 3 months before pregnancy - Weighted %" ELSE 0 END), 0) AS "Smoking Before Pregnancy %",
    ROUND(AVG(CASE WHEN "Any cigarette smoking during the last 3 months of pregnancy - Weighted %" IS NOT NULL THEN "Any cigarette smoking during the last 3 months of pregnancy - Weighted %" ELSE 0 END), 0) AS "Smoking During Pregnancy %",
    ROUND(AVG(CASE WHEN "Any cigarette smoking postpartum - Weighted %" IS NOT NULL THEN "Any cigarette smoking postpartum - Weighted %" ELSE 0 END), 0) AS "Smoking Postpartum %"
FROM
    Maternal_Indicators

GROUP BY
    Year


 * sqlite:///Maternal_Indicators.db
Done.


[(2016, 19.0, 9.0, 13.0),
 (2017, 19.0, 9.0, 13.0),
 (2018, 18.0, 9.0, 12.0),
 (2019, 17.0, 8.0, 11.0),
 (2020, 15.0, 7.0, 10.0),
 (2021, 13.0, 6.0, 8.0)]

- The reduction in smoking during pregnancy is notable

- On average, 52.56% of women who smoked before pregnancy stopped smoking during pregnancy.

- Of those who quit during pregnancy, approximately 35.46% resumed smoking postpartum.

### States with the highest and lowest smoking rates before pregnancy

In [18]:
%%sql 

SELECT
    "Site Name" AS State,
    COALESCE(ROUND(AVG(CASE WHEN "Any cigarette smoking during the 3 months before pregnancy - Weighted %" IS NOT NULL THEN "Any cigarette smoking during the 3 months before pregnancy - Weighted %" ELSE 0 END), 0), 0) AS "Smoking Before Pregnancy %"
FROM
    Maternal_Indicators
GROUP BY
    "Site Name"
ORDER BY
    "Smoking Before Pregnancy %" DESC;


 * sqlite:///Maternal_Indicators.db
Done.


[('West Virginia', 29.0),
 ('Kentucky', 28.0),
 ('Arkansas', 23.0),
 ('South Dakota', 22.0),
 ('Montana', 22.0),
 ('Missouri', 22.0),
 ('Maine', 22.0),
 ('Indiana', 22.0),
 ('Alaska', 22.0),
 ('Wyoming', 21.0),
 ('Vermont', 21.0),
 ('Oklahoma', 21.0),
 ('Alabama', 21.0),
 ('Pennsylvania', 20.0),
 ('North Dakota', 20.0),
 ('Mississippi', 20.0),
 ('Michigan', 20.0),
 ('Louisiana', 19.0),
 ('Iowa', 19.0),
 ('Tennessee', 18.0),
 ('New Hampshire', 18.0),
 ('Kansas', 18.0),
 ('Delaware', 18.0),
 ('Wisconsin', 17.0),
 ('North Carolina', 17.0),
 ('New York State', 16.0),
 ('Nebraska', 16.0),
 ('Virginia', 15.0),
 ('Oregon', 15.0),
 ('New Mexico', 15.0),
 ('Minnesota', 15.0),
 ('Illinois', 15.0),
 ('Texas', 14.0),
 ('Rhode Island', 14.0),
 ('Colorado', 14.0),
 ('Washington', 13.0),
 ('Hawaii', 12.0),
 ('Georgia', 12.0),
 ('Connecticut', 12.0),
 ('Massachusetts', 11.0),
 ('Maryland', 11.0),
 ('Florida', 11.0),
 ('Arizona', 11.0),
 ('New Jersey', 10.0),
 ('Utah', 8.0),
 ('District of Columbia', 8.0),
 ('Puerto Rico', 6.0)]